lib/ansible/modules/network/ios/ios_banner.py

```diff
@@ -0,0 +1,175 @@
+#!/usr/bin/python
+#
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+#
+
+ANSIBLE_METADATA = {'metadata_version': '1.0',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+
+DOCUMENTATION = """
+---
+module: ios_banner
+version_added: "2.3"
+author: "Ricardo Carrillo Cruz (@rcarrillocruz)"
+short_description: Manage multiline banners on Cisco IOS devices
+description:
+  - This will configure both login and motd banners on remote devices
+    running Cisco IOS.  It allows playbooks to add or remote
+    banner text from the active running configuration.
+extends_documentation_fragment: ios
+options:
+  banner:
+    description:
+      - The C(banner) argument specifies the banner that should be
+        configured on the remote device.  Current this module supports
+        configuration of either C(login) or C(motd) banners.
+    required: true
+    default: null
+  text:
+    description:
+      - The C(text) argument specifics the banner text that should be
+        present in the remote device running configuration.  This argument
+        accepts a multiline string.
+    required: false
+    default: null
+  state:
+    description:
+      - The C(state) argument specifies whether or not the configuration is
+        present in the current devices active running configuration.  When
+        this value is set to C(present), the configuration stanzas should be
+        in the current device configuration.  When this value is set to
+        C(absent), the configuration should not be in the current running
+        configuration.
+    required: false
+    default: present
+    choices: ['present', 'absent']
+"""
+
+EXAMPLES = """
+- name: configure the login banner
+  ios_banner:
+    banner: login
+    text: |
+      this is my login banner
+      that contains a multiline
+      string
+    state: present
+
+- name: remove the motd banner
+  ios_banner:
+    banner: motd
+    state: absent
+"""
+
+RETURN = """
+commands:
+  description: The list of configuration mode commands to send to the device
+  returned: always
+  type: list
+  sample:
+    - banner login
+    - this is my login banner
+    - that contains a multiline
+    - string
+    - EOF
+session_name:
+  description: The IOS config session name used to load the configuration
+  returned: always
```

Comment:
I think this is only returned on changed, again, diff against eos_banner in devel